custom_components/midea_ac_lan/config_flow.py

```diff
@@ -333,7 +335,9 @@ async def async_step_auto(
                 CONF_PORT: device.get(CONF_PORT),
                 CONF_MODEL: device.get(CONF_MODEL),
             }
-            if device_info := await self.cloud.get_device_info(device_id):
+            if self.cloud and (
```

Comment:
Because of the `if self.cloud is None:` above, the `if self.cloud and` here is not necessary

Reply:
Sorry, my bad. `get_midea_cloud` returns `MideaCloud | None`.
Maybe we should consider raise an error when `get_midea_cloud` still got `None`.

Reply:
The only way to get_midea_cloud returns None is when the server is invalid. This won´t happen due to the user selection of a valid value. Maybe the solution is to return the user to login step.

Reply:
Yes, actually `get_midea_cloud` won't return `None`, but the type includes `None`. So simply raise an error to pass mypy check is ok.

Reply:
Is not correct to raise an error just to pass mypy.
Better to fix the logic
If None should never return because the list comes from the library itself, let's fix the return type ;-)